pkg/app/piped/notifier/slack.go

```diff
@@ -112,7 +132,19 @@ func (s *slack) sendEvent(ctx context.Context, event model.NotificationEvent) {
 	}
 	if len(s.config.OAuthToken) != 0 {
 		if err := s.sendMessageViaAPI(ctx, msg); err != nil {
-			s.logger.Error(fmt.Sprintf("unable to send notification to slack: %v", err))
+			s.logger.Error(fmt.Sprintf("unable to send notification to slack using oauthToken: %v", err))
+		}
+		return
+	}
+	if len(s.config.OAuthTokenData) != 0 {
+		if err := s.sendMessageViaAPI(ctx, msg); err != nil {
+			s.logger.Error(fmt.Sprintf("unable to send notification to slack using oauthTokenData: %v", err))
+		}
+		return
+	}
+	if len(s.config.OAuthTokenFile) != 0 {
+		if err := s.sendMessageViaAPI(ctx, msg); err != nil {
+			s.logger.Error(fmt.Sprintf("unable to send notification to slack using oauthTokenFile: %v", err))
 		}
 		return
 	}
```

Comment:
```go
	if err := s.sendMessageViaAPI(ctx, msg); err != nil {
		s.logger.Error(fmt.Sprintf("unable to send notification to slack: %v", err))
	}
	return
```
I guess this is enough; we don't need to separate the error message here since the `err` printed out contains what is missing/failed already.

Reply:
um right, it's already

Reply:
you are right, updated